### PR TITLE
nixos/lxd: explicitly load kernel modules

### PR DESCRIPTION
**Pull request by ryneeverett**
###### Motivation for this change

This is analogous to #70447 and #76487.

These are all needed to attach a container to the default bridge
network, without which the final line of the following script fails with
the error for each respective kernel module listed below.

```sh
lxc storage create foo dir
lxc launch -s foo ubuntu:trusty bar
lxc network attach lxdbr0 bar
```

veth
----

> Error: Failed to start device 'lxdbr0': Failed to create the veth interfaces vethefbc3cd6 and vetha4abbcbc: Failed to run: ip link add dev vethefbc3cd6 type veth peer name vetha4abbcbc: RTNETLINK answers: Operation not supported

iptable_mangle
--------------

> lvl=eror msg="Failed to bring up network" err="Failed to list ipv4 rules for LXD network lxdbr0 (table mangle)" name=lxdbr0

xt_comment
----------

> lvl=error msg="Failed to bring up network" err="Failed to run: iptables -w -t filter -I INPUT -i lxdbr0 -p udp --dport 67 -j ACCEPT -m comment --comment generated for LXD network lxdbr0: iptables v1.8.4 (legacy): Couldn't load match `comment':No such file or directory\n\nTry `iptables -h' or 'iptables --help' for more information." name=lxdbr0

xt_MASQUERADE
-------------

> vl=eror msg="Failed to bring up network" err="Failed to run: iptables -w -t nat -I POSTROUTING -s 10.0.107.0/24 ! -d 10.0.107.0/24 -j MASQUERADE -m comment --comment generated for LXD network lxdbr0: iptables v1.8.4 (legacy): Couldn't load target `MASQUERADE':No such file or directory\n\nTry `iptables -h' or 'iptables --help' for more information." name=lxdbr0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Patryk27** [comment]
Just out of curiosity: should we load `iptables_mangle` even when `config.networking.nftables.enable` is active?

**Comment by ryneeverett** [comment]
> Just out of curiosity: should we load `iptables_mangle` even when `config.networking.nftables.enable` is active?

Good question. The answer is no, because iptables_mangle depends on ip_tables which is incompatible with nftables. (It appears that the nftables module has a hard check against ip_tables being loaded.) I pushed a fix.

**Comment by Mic92** [comment]
@GrahamcOfBorg test lxd

File: pkgs/tools/admin/lxd/default.nix [hunk 1]
```diff
@@ -58,6 +59,8 @@ buildGoPackage rec {
     installShellCompletion --bash go/src/github.com/lxc/lxd/scripts/bash/lxd-client
   '';
 
+  passthru.tests.lxd = nixosTests.lxd;
```
Comment (Patryk27) [inline pkgs/tools/admin/lxd/default.nix:62]: Out of curiosity, what does it do?

Comment (Mic92) [reply pkgs/tools/admin/lxd/default.nix:62]: It allows third-party tooling like ofborg discover nixos tests and build them.

**Comment by Mic92** [comment]
Probably unrelated but the nixos lxd test is broken right now: https://logs.nix.ci/?key=nixos/nixpkgs.97692&attempt_id=a0c03147-7108-49f6-bc06-1df04da57523

**Comment by Mic92** [comment]
@Patryk27 Do you have insights why the test fails?

**Comment by Patryk27** [comment]
@Mic92 not sure yet, I'm analyzing it now

**Comment by Patryk27** [comment]
@Mic92 seems like there's a race condition between `lxc config test ...` and our assertions, which makes the tests non-deterministic - adding `lxc restart test` in-between fixes this inconvenience:

```patch
diff --git a/nixos/tests/lxd.nix b/nixos/tests/lxd.nix
index d1e64238..fc549aed 100644
--- a/nixos/tests/lxd.nix
+++ b/nixos/tests/lxd.nix
@@ -96,6 +96,7 @@ in {
         ## limits.cpu ##

         machine.succeed("lxc config set test limits.cpu 1")
+        machine.succeed("lxc restart test")

         # Since Alpine doesn't have `nproc` pre-installed, we've gotta resort
         # to the primal methods
@@ -115,6 +116,7 @@ in {
         ## limits.memory ##

         machine.succeed("lxc config set test limits.memory 64MB")
+        machine.succeed("lxc restart test")

         assert (
             "MemTotal:          62500 kB"
```

**Comment by Mic92** [comment]
@GrahamcOfBorg test lxd

**Review by Patryk27 — APPROVED** [review]

**Review by mkg20001 — COMMENTED** [review]
LGTM

**Comment by mkg20001** [comment]
Any objections against merging this? Otherwise I'd merge

**Comment by mkg20001** [comment]
Ah, just noticed. The merge conflict got resolved wrongly, it's using pkgconfig, not pkg-config.

**Comment by mkg20001** [comment]
Rebased and fixed the error

**Pull request merged**